lib/puppet/module.rb

```diff
@@ -456,7 +456,11 @@ def subpath(type)
 
   def assert_validity
     if !Puppet::Module.is_module_directory_name?(@name) && !Puppet::Module.is_module_namespaced_name?(@name)
-      raise InvalidName, "Invalid module name #{@name}; module names must be alphanumeric (plus '-'), not '#{@name}'"
+      raise InvalidName, _(<<-ERROR_STRING).chomp % { name: @name }
+        Invalid module name '#{@name}'; module names must match either:
```

Comment:
Oh, wait, this isn't the right syntax. The variable here needs to be formatted like `%{name}`